### PR TITLE
Add multipart test cases for coldstorage.

### PR DESCRIPTION
**Pull request by sunfengyun**

### CODE REVIEW

File: coldstorage/test/coldstorage_test.go [hunk 1]
```diff
@@ -32,41 +35,104 @@ type InventoryRetrieval struct {
...
-	gcli := NewGlacierClient("XXX:XX", "XXX", "XXX", "XXX")
-	err := gcli.CreateVault("-", "VaultTest")
+func createSmallFile(content string, filename string) {
+	f, err := os.Create(filename)
 	if err != nil {
-		t.Fatal("Create vault fail:", err)
-	} else {
-		t.Log("Create vault success!")
+		Logger.Println(5, "With error: ", err)
+	}
```
Comment (sunfch) [inline coldstorage/test/coldstorage_test.go]: if failed to create file, the test case should be not continued

File: coldstorage/test/coldstorage_test.go [hunk 2]
```diff
@@ -32,41 +35,104 @@ type InventoryRetrieval struct {
...
-	} else {
-		content := "This file is used for cold storage function test."
-		_, err = f.Write([]byte(content))
+		Logger.Println(5, "With error: ", err)
+	}
+	f, err = os.OpenFile(filename, os.O_CREATE|os.O_RDWR, os.ModeAppend|os.ModeSetuid)
+	if err != nil {
+		Logger.Println(5, "With error: ", err)
+	}
+	count := math.Ceil(float64(size) / 1000)
```
Comment (sunfch) [inline coldstorage/test/coldstorage_test.go]: Converting "size" to fload64 is not necessary.

File: coldstorage/test/coldstorage_test.go [hunk 3]
```diff
@@ -32,41 +35,104 @@ type InventoryRetrieval struct {
...
+	count_64 := int64(int(count))
```
Comment (sunfch) [inline coldstorage/test/coldstorage_test.go]: the statement "int64(int(count))"  is strange

File: coldstorage/test/coldstorage_test.go [hunk 4]
```diff
@@ -32,41 +35,104 @@ type InventoryRetrieval struct {
...
-	_, err = file.Read(buffer)
+	num := int(math.Ceil(float64(fileInfo.Size()) / float64(chunkSize)))
+	fi, err := os.OpenFile(filename, os.O_RDONLY, os.ModePerm)
 	if err != nil {
-		t.Fatal("Read file content to buffer fail:", err)
+		Logger.Println(5, "With error: ", err)
+	}
+	b := make([]byte, chunkSize)
+	var i int64 = 1
+	for ; i <= int64(num); i++ {
```
Comment (sunfch) [inline coldstorage/test/coldstorage_test.go]: why not start from 0

File: coldstorage/test/coldstorage_test.go [hunk 5]
```diff
@@ -32,41 +35,104 @@ type InventoryRetrieval struct {
...
-func TestColdStorage(t *testing.T) {
-	gcli := NewGlacierClient("XXX:XX", "XXX", "XXX", "XXX")
-	err := gcli.CreateVault("-", "VaultTest")
+func createSmallFile(content string, filename string) {
+	f, err := os.Create(filename)
 	if err != nil {
-		t.Fatal("Create vault fail:", err)
-	} else {
-		t.Log("Create vault success!")
+		Logger.Println(5, "With error: ", err)
```
Comment (brucen1030) [inline coldstorage/test/coldstorage_test.go]: It's suggested to use t.Log in test file.

File: coldstorage/test/coldstorage_test.go [hunk 6]
```diff
@@ -32,41 +35,104 @@ type InventoryRetrieval struct {
...
+
+	err := gcli.CreateVault("-", "VaultTest")
+	if err != nil {
+		t.Fatal("Create vault fail:", err)
 	} else {
-		fileBytes = bytes.NewReader(buffer) // Converted to io.ReadSeeker type.
+		t.Log("Create vault success!")
+	}
+
+	content := "This file is used for cold storage single archive function test."
```
Comment (brucen1030) [inline coldstorage/test/coldstorage_test.go]: A single string for small file test should be enough, just like test/go/Test_PutXXX, or aws-sdk-go example service/glacier/examples_test.c as below.

func ExampleGlacier_UploadArchive_shared00() {
	svc := glacier.New(session.New())
	input := &glacier.UploadArchiveInput{
		AccountId:          aws.String("-"),
		ArchiveDescription: aws.String(""),
		Body:               aws.ReadSeekCloser(strings.NewReader("example-data-to-upload")),
		Checksum:           aws.String(""),
		VaultName:          aws.String("my-vault"),
	}

File: coldstorage/test/coldstorage_test.go [hunk 7]
```diff
@@ -110,28 +176,103 @@ func TestColdStorage(t *testing.T) {
 	archivebuf := new(bytes.Buffer)
 	_, err = archivebuf.ReadFrom(ioReadCloser)
 	if err != nil {
-		t.Fatal("ReadFrom(ioReadCloser) fail:", err)
-	} else {
-		s = archivebuf.String()
+		t.Fatal("ReadFromioReadCloser fail:", err)
```
Comment (brucen1030) [inline coldstorage/test/coldstorage_test.go]: And it's not necessary to for sleep.
Sleep for seconds should be enough for Glacier.

File: coldstorage/test/coldstorage_test.go [hunk 8]
```diff
@@ -110,28 +176,103 @@ func TestColdStorage(t *testing.T) {
...
+	if err != nil {
+		t.Fatal("CreateMultipart fail:", err)
+	} else {
+		t.Log("CreateMultipart success!")
+	}
+
+	var bytesize = 25690112
+	var chunkSize int64 = 8388608
+	createBigFile(bytesize, "BigFile.txt")
+	num := splitBigFile("BigFile.txt", chunkSize)
```
Comment (brucen1030) [inline coldstorage/test/coldstorage_test.go]: For big file, i think it's better to write a reader to produce 1M data, or whatever size you like, for upload.

**Pull request closed without merging**